sormas-backend/src/main/java/de/symeda/sormas/backend/immunization/ImmunizationFacadeEjb.java

```diff
@@ -646,6 +649,28 @@ public void syncSharesAsync(Immunization immunization) {
 		}, 5, TimeUnit.SECONDS);
 	}
 
+	public void copyImmunizationsToLeadPerson(ImmunizationDto immunizationDto, PersonDto leadPerson) {
+		Immunization newImmunization = new Immunization();
+		newImmunization.setUuid(DataHelper.createUuid());
+		newImmunization = fillOrBuildEntity(immunizationDto, newImmunization, false, true);
+
+		newImmunization.setPerson(personService.getByReferenceDto(leadPerson.toReference()));
+
+		List<Vaccination> vaccinationEntities = new ArrayList<>();
```

Comment:
You could move the generation of `vaccinationentities` to `VaccinationFacade`. In this way `fillOrBuildEntity` can remain private.

Reply:
recommendation implemented